src/pieces/assets/assets_api.py

```diff
@@ -0,0 +1,235 @@
+					fragment=SeededFragment(
+						string=TransferableString(raw=raw_string)
+					)
+				),
+				metadata=metadata  # This should be constructed as per the SDK's definition
+			),
+			type="SEEDED_ASSET"
+		)
+		
+		# Creating the new asset using the assets API
+		created_asset = assets_api.assets_create_new_asset(transferables=False, seed=seed)
+		return created_asset
+	
+	@classmethod
+	def get_assets_snapshot(cls):
+		if cls.assets_snapshot:
+			return cls.assets_snapshot
+	
+
+		assets_api = AssetsApi(Settings.api_client)
+		# Call the API to get assets identifiers
+		api_response = assets_api.assets_identifiers_snapshot()
+
+		# Extract the 'id' values from each item in the 'iterable' list
+		cls.assets_snapshot = {item.id:None for item in api_response.iterable}
+
+		return cls.assets_snapshot
+		
+	@classmethod
+	def update_asset_snapshot(cls,asset_id):
+		asset_api = AssetApi(Settings.api_client)
+		asset = asset_api.asset_snapshot(asset_id)
+		cls.assets_snapshot[asset_id] = asset # Cache the asset
+		return asset
+
+	@classmethod
+	def worker(cls):
+		try:
+			while True:
+				asset_id = cls.asset_queue.get(block=cls.block,timeout=5)
+				cls.asset_set.remove(asset_id)  # Remove asset_id from the set
+				cls.update_asset_snapshot(asset_id)
+				cls.asset_queue.task_done()
+		except queue.Empty: # queue is empty and the block is false
+			if cls.block:
+				cls.worker() # if there is more assets to load
+			return # End the worker
+	
+	@classmethod
+	def assets_snapshot_callback(cls,ids:StreamedIdentifiers):
+		# Start the worker thread if it's not running
+		cls.worker()
+		cls.block = True
+		for item in ids.iterable:
+			asset_id = item.asset.id
+			if asset_id not in cls.assets_snapshot:
+				cls.assets_snapshot[asset_id] = None
+			
+			if asset_id not in cls.asset_set:
+				if item.deleted:
+					cls.assets_snapshot.pop(asset_id)
+				else:
+					cls.asset_queue.put(asset_id)  # Add asset_id to the queue
+					cls.asset_set.add(asset_id)  # Add asset_id to the set
+		cls.block = False # Remove the block to end the thread
+
+
+
+	@classmethod
+	def get_asset_snapshot(cls,asset_id:str):
+		asset = cls.assets_snapshot.get(asset_id)
+		if asset:
+			return asset
+		else:
+			return cls.update_asset_snapshot(asset_id)
+
+	
+
+	@classmethod
+	def edit_asset_name(cls,asset_id, new_name):
+		asset_api = AssetApi(Settings.api_client)
+
+		# Get the asset using the provided asset_id
+		asset = cls.get_asset_snapshot(asset_id)
+
+		# Check if the existing name is found and update it
+		existing_name = asset.name
+		asset.name = new_name
+		print(f"Asset name changed from '{existing_name}' to '{new_name}'")
+
+		# Update the asset using the API
+		try:
+			asset_api.asset_update(asset=asset, transferables=False)
+			print("Asset name updated successfully.")
+		except Exception as e:
+			show_error("Error updating asset: ",{e})
+
+
+	@staticmethod
+	def delete_asset_by_id(asset_id):
+		delete_instance = AssetsApi(Settings.api_client)
```

Comment:
let's add a github issue to make a central class that holds all the api instances so we don't need to create a new instance like this each time we make an api call. kind of like this file: https://github.com/pieces-app/copilot-ts/blob/main/src/controllers/ConnectorSingleton.ts

Reply:
Yeah thank you good idea!

Reply:
added!